*: add libgeos to more places

otan opened the pull request:
See individual commits for more details.

Resolves #61665

otan committed:
logictestccl: mark as working under bazel

Bump size due to timing time, and add GEOS.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62387)
<!-- Reviewable:end -->

RaduBerinde commented:
Can you add it to these tests as well?

memo.TestStats
norm.TestNormRules
xform.TestExternal
xform.TestRules

I'm fixing them in #62109 but they still fail because of libgeos.

otan commented:
done!

rickystewart commented on pkg/sql/BUILD.bazel:
```diff
@@ -489,7 +489,9 @@ go_test(
     ],
-    data = glob(["testdata/**"]),
+    data = glob(["testdata/**"]) + [
+        "@cockroach//c-deps:libgeos",
```
Why was this necessary here?

otan replied:
the sql logic tests will depend on this

otan committed:
*: add libgeos to sql,execbuilder,norm,xnorm,memo BUILD.bazel

The tests still fail for different reasons, but this is one less reason
to.

Release note: None

otan commented:
bors r=rickystewart
